build/templates/rep_caps.rst.mako

```diff
@@ -19,16 +19,20 @@
 
 ${helper.get_rst_header_snippet('Repeated Capabilities', '=')}
 
-    Repeated capabilities attributes are used to set the `channel_string` parameter to the
-    underlying driver function call. This can be the actual function based on the :py:class:`Session`
-    method being called, or it can be the appropriate Get/Set Attribute function, such as :c:`${config['c_function_prefix']}SetAttributeViInt32()`.
+    :py:class:`${module_name}.Session` supports "Repeated Capabilities", which are multiple instances of the same type of
+    functionality. The repeated capabilities supported by :py:class:`${module_name}.Session` are:
 
-    Repeated capabilities attributes use the indexing operator :python:`[]` to indicate the repeated capabilities.
-    The parameter can be a string, list, tuple, or slice (range). Each element of those can be a string or
-    an integer. If it is a string, you can indicate a range using the same format as the driver: :python:`'0-2'` or
-    :python:`'0:2'`
+% for rep_cap in config['repeated_capabilities']:
+<%
+name = rep_cap['python_name']
+%>\
+    #. ${name}_
+% endfor
+
+    Use the indexing operator :python:`[]` to indicate which repeated capability you are trying to access.
+    The parameter can be an integer, a string, a list, a tuple, or slice (range).
 
-    Some repeated capabilities use a prefix before the number and this is optional
+    The recommended way of accessing repeated capabilities is with an integer :python:`[0]` or range :python:`[0:2]`.
```

Comment:
> range :python:`[0:2]`

I don't think this will work since this is not the correct syntax for Python `range`. But `['0:2']` will work because of the custom logic we have for handling range-based strings.

Having said that, I don't think any of the ways I've enumerated above can be recommended over others. `int` is obviously the go-to for single instances. Basic sequence types and `slice` are good to use because they're part of the language, Range-based strings provides syntactic sugar for defining ranges. e.g. `['0:2']` rather than `[range(0,2)]`.

Reply:
I see @marcoskirsch doesn't like us to recommend the range-based strings. Since I saw we had logic specifically for handling it, I thought it was important. But it sounds like it was ill-conceived. I don't mind not documenting it.

Once we land on recommended ways, we should go over the examples to ensure only recommended types are used.